docs/source/api/v2/servers_hostname_update_status.rst

```diff
@@ -52,16 +52,26 @@ Response Structure
 ------------------
 Each object in the returned array\ [1]_ will contain the following fields:
 
+:configUpdateTime:     The last time an update was requested for this server. This field defaults to standard epoch
+:configApplyTime:      The last time an update was applied for this server. This field defaults to standard epoch
```

Comment:
I thought adding was _ok_ since it doesn't break any backwards compatibility. Are previous API versions then locked? Does this, then, also hold true for previous go client versions since they are essentially 1:1 with the API versions?

Reply:
> Are previous API versions then locked? 

Yes.

> Does this, then, also hold true for previous go client versions since they are essentially 1:1 with the API versions?

Yes and no.

The API follows semantic versioning. All changes must be reflected in a version change. Backwards-compatible changes necessitate a minor version bump while breaking changes necessitate a major version bump. Since we already have 4.0 - and since T3C already links against that unstable version - there's no real reason to create and maintain a 3.2 version, so while the changes only _necessitate_ a minor version bump it's easier to include them in the major version.

The client promises only compile-time compatibility. What this essentially means is that when we made API version 3.1, for example, we didn't make a `github.com/apache/trafficcontrol/traffic_ops/v3-1-client`, instead our library returns type alias structures like e.g. `UserV3` which is an alias of `UserV30` that was changed to be an alias to the expanded `UserV31`. So we do minor version bumps to the client behind the scenes - but with 4.0 being made, we _did_ create a `github.com/apache/trafficcontrol/traffic_ops/v4-client`. Those are different import paths, so all breaking changes are taken on knowingly and intentionally by users of the client. So code written using our client for 3.0 will work even when that's _actually_ hitting 3.1, but you'll never see breaking changes to the call signatures (and the client has functionality to fall back to previous minor versions such that if e.g. APIv3.1 isn't available it'll try APIv3.0 before giving up).

Reply:
I've removed the new fields from the previous API versions.